shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -24,6 +35,94 @@ PlatformChannel::PlatformChannel(flutter::BinaryMessenger* messenger)
 
 PlatformChannel::~PlatformChannel() {}
 
+#if defined(MOBILE) || defined(WEARABLE)
+namespace {
+
+class FeedbackManager {
+ public:
+  enum class ResultCode {
+    OK,
+    NOT_SUPPORTED_ERROR,
+    PERMISSION_DENIED_ERROR,
+    UNKNOWN_ERROR
+  };
+
+  FeedbackManager() {
+    FT_LOGD("Enter FeedbackManager::FeedbackManager()");
+
+    auto ret = feedback_initialize();
+    if (FEEDBACK_ERROR_NONE != ret) {
+      FT_LOGD("feedback_initialize() failed with error: [%d] (%s)", ret,
+              get_error_message(ret));
+      return;
+    }
+    FT_LOGD("feedback_initialize() succeeded");
+    properly_initialized_ = true;
+
+    ret = feedback_is_supported_pattern(
+        FEEDBACK_TYPE_VIBRATION, FEEDBACK_PATTERN_SIP, &vibration_supported_);
+    if (FEEDBACK_ERROR_NONE != ret) {
+      FT_LOGD("feedback_is_supported_pattern() failed with error: [%d] (%s)",
+              ret, get_error_message(ret));
+      return;
+    }
+    FT_LOGD("feedback_is_supported_pattern() succeeded");
+  }
+
+  ~FeedbackManager() {
+    FT_LOGD("Enter FeedbackManager::~FeedbackManager");
+
+    if (!properly_initialized_) {
+      return;
+    }
+
+    auto ret = feedback_deinitialize();
+    if (FEEDBACK_ERROR_NONE != ret) {
+      FT_LOGD("feedback_deinitialize() failed with error: [%d] (%s)", ret,
+              get_error_message(ret));
+      return;
+    }
+    FT_LOGD("feedback_deinitialize() succeeded");
+  }
+
+  ResultCode Vibrate() {
+    FT_LOGD("Enter FeedbackManager::Vibrate()");
+
+    if (!properly_initialized_) {
+      FT_LOGD(
+          "Cannot run Vibrate(): FeedbackManager.properly_initialized_ is "
+          "false");
+      return ResultCode::UNKNOWN_ERROR;
+    }
+
+    if (!vibration_supported_) {
+      FT_LOGD("HapticFeedback.Vibrate() is not supported");
+      return ResultCode::NOT_SUPPORTED_ERROR;
+    }
+
+    auto ret =
+        feedback_play_type(FEEDBACK_TYPE_VIBRATION, FEEDBACK_PATTERN_SIP);
+    FT_LOGD("feedback_play_type() returned: [%d] (%s)", ret,
+            get_error_message(ret));
+    if (FEEDBACK_ERROR_NONE == ret) {
+      return ResultCode::OK;
+    } else if (FEEDBACK_ERROR_PERMISSION_DENIED == ret) {
```

Comment:
is this statement necessary, if it is mapped to the UNKNOWN_ERROR which is also a default value in line 114?

Reply:
or maybe line 110 should be changed to FeedbackManager::ResultCode::PERMISSION_DENIED_ERROR which is checked in line 151?

Reply:
This is a mistake - it should return `FeedbackManager::ResultCode::PERMISSION_DENIED_ERROR `. I'll fix it